### PR TITLE
Task5

### PR DESCRIPTION
**Pull request by Neverhoodof**

### CODE REVIEW

**Review by Zawadskiy — CHANGES_REQUESTED** [review]

File: src/main/java/com/walking/intensive/chapter1/task5/Task5.java [hunk 1]
```diff
@@ -1,128 +1,143 @@
...
- * треугольника окружностей является центральной в Геометрии.
- *
- * <p>Реализуйте представленные ниже методы в соответствии с заданными условиями.
- *
- * <p><a href="https://github.com/KFalcon2022/intensive-tasks-2024/blob/master/README.md">Требования к оформлению</a>
- */
+import java.util.Arrays;
+
 public class Task5 {
     public static void main(String[] args) {
```
[the remark this answers is not in the document]
Comment (Neverhoodof) [reply src/main/java/com/walking/intensive/chapter1/task5/Task5.java:6]: Не понимаю как провести Checkstyle

[the remark this answers is not in the document]
Comment (Neverhoodof) [reply src/main/java/com/walking/intensive/chapter1/task5/Task5.java:6]: Хм, нашел вроде

File: src/main/java/com/walking/intensive/chapter1/task5/Task5.java [hunk 2]
```diff
@@ -1,128 +1,134 @@
...
-        //        Место для вашего кода
 
-        return 0; // Заглушка. При реализации - удалить
+        if (!isDegenerate(a, b, c)) {
+            return -1;
+        }
+
+        double semiPerimeter = (a + b + c) / 2;
+
+        return (a * b * c) / (4 * Math.sqrt(semiPerimeter * (semiPerimeter - a) * (semiPerimeter - b) * (semiPerimeter - c)));
```
Comment (Zawadskiy) [inline src/main/java/com/walking/intensive/chapter1/task5/Task5.java:120]: Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter1/task5/Task5.java:120: Line is longer than 120 characters (found 126). [LineLength]

**Review by Zawadskiy — CHANGES_REQUESTED** [review]

File: src/main/java/com/walking/intensive/chapter1/task5/Task5.java [hunk 3]
```diff
@@ -1,128 +1,135 @@
...
-     * <p>Если входные данные некорректны - метод должен возвращать -1.
-     */
     static double getCircumradius(double a, double b, double c) {
-        //        Место для вашего кода
 
-        return 0; // Заглушка. При реализации - удалить
+        if (!isDegenerate(a, b, c)) {
+            return -1;
+        }
+
```
Comment (Zawadskiy) [inline src/main/java/com/walking/intensive/chapter1/task5/Task5.java:117]: > Task :checkstyleMain
Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter1/task5/Task5.java:117: Line has trailing spaces. [RegexpSingleline]

**Review by Zawadskiy — APPROVED** [review]

**Pull request closed without merging**